clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -579,7 +579,9 @@
      * either strings or objects of the appropriate type (for example a numeric configuration would accept either the
      * string "42" or the integer 42).
      * <p>
-     * Valid configuration strings are documented at {@link ConsumerConfig}
+     * Valid configuration strings are documented at {@link ConsumerConfig}.
+     * <p>
+     * Note: after using the {@code KafkaConsumer} you must always {@link #close()} it to avoid resource leaks.
```

Comment:
I wonder if it makes sense to simplify this and say that 'after creating a {@code KafkaConsumer} you must always call close'. 

This is not strictly true, but would make it consistent with the streams and producer javadoc. If we want to be specific, then we should go the whole way and say 'after calling poll, we should always close'.